google-cloud-logging/src/main/java/com/google/cloud/logging/ResourceTypeEnvironmentGetter.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging;
+
+import com.google.cloud.MetadataConfig;
+
+public interface ResourceTypeEnvironmentGetter {
+
+  /**
+   * Gets the value of the specified environment variable.
+   *
+   * @param name the name of the environment variable
+   * @return the string value of the variable, or <code>null</code> if the variable is not defined
+   *     in the system environment
+   * @see System#getenv()
+   */
+  String getEnv(String name);
+
+  /**
+   * Gets the value of the specified metadata server attribute.
+   *
+   * @param name the name of the metadata server attribute.
+   * @return the string value of the attribute, or <code>null</code> if the attribute is not defined
+   *     in the metadata or the server is not available.
+   * @see MetadataConfig#getAttribute()
+   */
+  String getAttribute(String name);
+}
+
+final class ResourceTypeEnvironmentGetterImpl implements ResourceTypeEnvironmentGetter {
+
+  @Override
+  public String getEnv(String name) {
+    return System.getenv(name);
```

Comment:
Respectfully disagree. The goal of this wrapper is only to allow successful unit testing. Returning `null` is both handled in the code and validated in the tests. Throwing exception can happened in a case of the illegal input which is unit tested.

Reply:
The problem is that the code might change hands and evolve. It might be executed in customer environment and throw exceptions causing customer outages. This might be undetected in our tests (e.g. throwing security exception on env variable reading). Our library code needs to be implemented "defensively", so the library is not crashing.

Reply:
Got it, I will add handling of `SecurityException`. `NullPointerException` should not be thrown by our code in any circumstances and if it happens it should be handled by our tests or worst case cause the crash so we can fix it.